src/lib/server/links/utils.ts

```diff
@@ -1,15 +1,76 @@
 import QRCode from 'qrcode';
-import { LINKS } from './data';
+import type { Link, ParsedLink } from './types';
 
-export function parseLinkId(link: string): keyof typeof LINKS | undefined {
-  const cleanedLinkName = link
+// parseLink parses the incoming link from a request.
```

Comment:
had to read the source and tests to figure out what this do.
better comment probably something like "given a link, return the relative path name, query string, and hash", in which case how come not use built in url parsing https://developer.mozilla.org/en-US/docs/Web/API/URL/URL works on all major browsers, deno, and node

Reply:
I gave a shot at updating the content of the comment to be a bit more clear <https://github.com/EthanThatOneKid/acmcsuf.com/pull/607/commits/0403e9899046a9e746703e1d5731eab5d8007720>.

Reply:
> how come not use built in url parsing https://developer.mozilla.org/en-US/docs/Web/API/URL/URL works on all major browsers, deno, and node

As far as I know, the `new URL()` constructor will throw an error if the string is not a valid URL. The "link" being passed into `parseLinkId` is expected to be the absolute pathname of the already parsed URL.
<https://github.com/EthanThatOneKid/acmcsuf.com/blob/697014c6ae2d4d6c4a095ab8b9f9d95e9e515adb/src/lib/server/links/shortener.ts#L7-L8>

This makes me realize that perhaps the current argument name perhaps is not sufficient. Perhaps `pathname` or `absolutePathname` instead of `url`. What do you think, @jaasonw?

Reply:
yea sure

Reply:
We ended up leveraging URL parsing in the latest version of this PR. Good suggestion!